src/components/App.tsx

```diff
@@ -9,35 +9,75 @@ import { store, persistor } from "../store";
 import { refreshMembers } from "../store/actions/members";
 import { MessagingManager } from "./MessagingManager";
 import { DropdownWrapper } from "./DropdownWrapper";
+import { Loading } from "./shared/Loading";
+import { AppState } from "react-native";
 
 // refresh the members/operations on app start
 const onBeforeLift = async () => {
   await refreshMembers()(store.dispatch, store.getState, undefined);
 };
 
-const AppRoot: React.StatelessComponent = () => {
-  return (
-    // TODO: see if there's a better way to make thunk types work properly aside
-    // from cast to `any`
-    <Provider store={store as any}>
-      {/* TODO: decide if we should show a loading indicator here. */}
-      <PersistGate
-        loading={null}
-        persistor={persistor}
-        onBeforeLift={onBeforeLift}
-      >
-        <AuthManager>
-          <MessagingManager>
-            <DropdownWrapper>
-              <Navigation />
-            </DropdownWrapper>
-          </MessagingManager>
-        </AuthManager>
-      </PersistGate>
-    </Provider>
-  );
+type AppRootProps = {};
+
+type AppRootState = {
+  shouldRefreshOnNextForeground: boolean;
 };
 
+class AppRoot extends React.Component<AppRootProps, AppRootState> {
+  constructor(props: AppRootProps) {
+    super(props);
+    this.state = {
+      // First refresh is from PersistGate#onBeforeLift
+      shouldRefreshOnNextForeground: false
+    };
+  }
+
+  componentDidMount() {
+    AppState.addEventListener("change", this._handleAppStateChange);
+  }
+
+  componentWillUnmount() {
+    AppState.removeEventListener("change", this._handleAppStateChange);
+  }
+
+  _handleAppStateChange = (nextAppState: string) => {
+    if (nextAppState === "background") {
```

Comment:
Hmm, could we just compare the `nextAppState` value to `AppState.currentState` instead of maintaining the `shouldRefreshOnNextForeground` state ourselves?

i.e. in the handler, and using the documentation at https://facebook.github.io/react-native/docs/appstate.html:
```
const currentState = AppState.currentState;
if (
  // prevents us from refreshing on app start
  currentState &&
  nextAppState === "foreground" &&
  nextAppState !== currentState
) {
  refreshMembers()
}
```

Reply:
hm, when I was testing, nextAppState === currentState...

Reply:
I was reading https://medium.freecodecamp.org/where-do-i-belong-a-guide-to-saving-react-component-data-in-state-store-static-and-this-c49b335e2a00 and it says that component state that does not impact rendering should be stored on the class directly (i.e. `this.shouldRefreshOnNextForeground`), as state changes will trigger a re-render. I think we're doing similar things elsewhere in the app, I wonder what the performance implication of that is.

Reply:
kk